prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -188,6 +189,41 @@ class DeliveryFunction(
         return this.getDetailedView(request, id)
     }
 
+    /**
+     * Endpoint for intermediary receivers to verify status of messages. It passes
+     * a null engine to the retrieveMetadata function because Azure gets upset if there
+     * are any non-annotated parameters in the method signature other than ExecutionContext
+     * and we needed the engine to be a parameter so it can be mocked for tests
+     *
+     */
+    @FunctionName("getEtorMetadataForDelivery")
+    fun getEtorMetadata(
+        @HttpTrigger(
+            name = "getEtorMetadataForDelivery",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{reportId}/delivery/etorMetadata"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("reportId") reportId: UUID,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        return this.retrieveETORIntermediaryMetadata(request, reportId, context, null)
+    }
+
+    /**
+     * Function for finding the associated report ID that the intermediary knows about given the report ID that the receiver is
+     * given from report stream
+     */
+    override fun getLookupId(reportId: UUID): UUID? {
+        // the delivery endpoint is called by the final receiver with a sent report ID, where TI
+        // knows about the related submission report ID
+
+        val reportGraph = ReportGraph(workflowEngine.db)
+        val root = reportGraph.getRootReport(reportId)
+
+        return root?.reportId
```

Comment:
This seems like an exceptional case that should be at least logged if not generate an error message right?  Isn't this where we delivered a report but then there is record of it getting submitted?

Reply:
Good point.  Fixed.